ui/src/app/components/workflows/workflows.component.ts

```diff
@@ -45,6 +47,27 @@ export class WorkflowsComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  isWorkflowHighlighted(id: number): boolean {
+    return this.router.url.split('/').some((part) => part === id.toString());
+  }
+
+  isProjectClosed(project: string, workflows: WorkflowIdentityModel[]): boolean {
+    const isOpen =
+      this.openedProjects.has(project) || workflows.some((workflow: WorkflowIdentityModel) => this.isWorkflowHighlighted(workflow.id));
+    if (isOpen && !this.openedProjects.has(project)) {
+      this.openedProjects.add(project);
+    }
+    return !isOpen;
```

Comment:
Here, you could then just return `this.openedProjects.has(project)`

Imo, the logic is a bit easier to follow than the detour over `isOpen`

Reply:
Should be !this.openedProjects.has(project). Fixed.